### PR TITLE
wgsl: Add validation tests for continue statement

### PR DESCRIPTION
**Pull request by jrprice**
Issue #1681

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/parse/continue.spec.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    pass: false,
+  },
+  compound_continue: {
+    src: '{ continue; }',
+    pass: false,
+  },
+  loop_continue: {
+    src: 'loop { if false { break; } continue; }',
+    pass: true,
+  },
```
Comment (dj2) [inline src/webgpu/shader/validation/parse/continue.spec.ts:21]: Continue with expression `loop { continue 2 + 3;}` fails

File: src/webgpu/shader/validation/parse/continue.spec.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  while_continue: {
+    src: 'while true { continue; }',
+    pass: true,
+  },
+  for_continue: {
+    src: 'for (;true;) { continue; }',
+    pass: true,
+  },
```
Comment (dj2) [inline src/webgpu/shader/validation/parse/continue.spec.ts:29]: `for(continue;;) {break;}`
`for(;continue;) {break;}`
`for(;;continue) {break;}`

File: src/webgpu/shader/validation/parse/continue.spec.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  loop_continue_before_decl_not_used_in_continuing: {
+    src: 'loop { continue; let cond = false; continuing { break if false; } }',
+    pass: true,
+  },
+  loop_nested_continue_before_decl_used_in_continuing: {
+    src: 'loop { if false { continue; } let cond = false; continuing { break if cond; } }',
+    pass: false,
+  },
+};
```
Comment (dj2) [inline src/webgpu/shader/validation/parse/continue.spec.ts:74]: continue at module scope

**Review by dj2 — COMMENTED** [review]
Few more possible tests.

File: src/webgpu/shader/validation/parse/continue.spec.ts (hunk 1 above)
Comment (jrprice) [reply src/webgpu/shader/validation/parse/continue.spec.ts:21]: Done.

File: src/webgpu/shader/validation/parse/continue.spec.ts (hunk 2 above)
Comment (jrprice) [reply src/webgpu/shader/validation/parse/continue.spec.ts:29]: Done.

File: src/webgpu/shader/validation/parse/continue.spec.ts (hunk 3 above)
Comment (jrprice) [reply src/webgpu/shader/validation/parse/continue.spec.ts:90]: Done.

**Review by dj2 — APPROVED** [review]

**Pull request merged**